api/src/pcapi/core/offers/validation.py

```diff
@@ -410,6 +414,17 @@ def check_offer_extra_data(subcategory_id: str, extra_data: dict | None) -> None
         raise api_errors
 
 
+def compute_ean_error(ean: str | None) -> str | None:
+    if ean is None:
+        return None
+
+    has_correct_length = len(ean) == 8 or len(ean) == 13
+    if ean.isdigit() and has_correct_length:
```

Comment:
programmation défensive : 
`if isinstance(ean, string) and ....`

Reply:
> extra_data["isbn"] est forcément une chaîne de caractères (ou None)

ça va être le cas pour notre front mais pas forcément pour les utilisateurs de l'api

si un utilisateur nous envoie l'ean `12345678` plutôt que `"12345678"` :

Reply:
À mon avis, l'API ne devrait accepter qu'une chaîne de caractères, **sans _cast_ automatique**. Comme tu le dis, ça paraît un peu optimiste de préfixer par un ou plusieurs zéros s'il manque des caractères.